### PR TITLE
Adds luxon DateTimes & uses them inside critical functions

### LINKED ISSUE #385: Scenario 2
### Acceptance Criteria

- [ ]

### PR DESCRIPTION
**Pull request by lomky**
PR 1 of 2

This is the minimal changes to actually solve #385, while leaving the rest of the date functionality unchanged.

 - Adds luxon DateTime
 - Sets our Luxon TZ to 'America/Los_Angeles'
 - Teaches `formatFromApiGateway` to act properly around midnight
 - Teaches `isDatePast` to act properly around midnight

However this is a _bad idea_ to do halfway! now we have two time-handling processes! That handle time zones differently! AHHH

So see the next PR in the stack for the full refactor. I just want to hold this distinct for reviewing.

Testing:
  - [ ] All scenarios still load properly - but especially 2 & 3
  - [ ] Tests pass, with no snapshot changes

===

Resolves #385

- [x] Scenario 2 should be returned for appointments that are "today" until 11:59pm PT. Scenario 3 should not be displayed until it is 12:00am PT the next calendar day
- [ ] Changes are tested on Staging post-merge into `main`

### CODE REVIEW

**Comment by lomky** [comment]
The tests are failing because I didn't update the tests until to follow on - if we decide to merge this one without the other I'll backport it

File: utils/formatDate.ts [hunk 1]
```diff
@@ -16,12 +16,15 @@ import { isValid } from 'date-fns'
...
 import enUS from 'date-fns/locale/en-US'
 import es from 'date-fns/locale/es'
+import { DateTime, Settings } from 'luxon'
 
 import { ApiGatewayDateString } from '../types/common'
 
 const pacificTimeZone = 'America/Los_Angeles'
 const apiGatewayFormat = "yyyy-MM-dd'T'HH:mm:ss"
 
+Settings.defaultZone = 'America/Los_Angeles'
```
Comment (rocketnova) [inline utils/formatDate.ts:26]: Nit:

```suggestion
Settings.defaultZone = pacificTimeZone
```

File: utils/formatDate.ts [hunk 2]
```diff
@@ -48,16 +51,12 @@ export function parseApiGatewayDate(dateString: ApiGatewayDateString): Date {
...
- * currently PST (-8) or PDT (-7).
  */
-export function formatFromApiGateway(daysOffset = 1): string {
-  const today = new Date()
-  today.setDate(today.getDate() + daysOffset)
-  today.setUTCHours(8, 0, 0, 0)
-  return format(today, apiGatewayFormat)
+export function formatFromApiGateway(daysOffset = 1, today = DateTime.now()): string {
+  const newDate = today.plus({ days: daysOffset }).startOf('day')
+  return newDate.toFormat(apiGatewayFormat)
```
Comment (rocketnova) [inline utils/formatDate.ts:59]: 🎉

File: tests/utils/formatDate.test.ts [hunk 3]
```diff
@@ -88,3 +95,28 @@ describe('Formatting dates', () => {
     expect(formatted).toEqual('9/27/2013')
   })
 })
+
+// Test formatFromApiGateway()
+describe('Requesting a date adjusted from existing in the correct format', () => {
+  it('has our time set to PT!', () => {
+    expect(DateTime.local().zoneName).toEqual('America/Los_Angeles')
```
Comment (rocketnova) [inline tests/utils/formatDate.test.ts:102]: Question about luxon: Do you need to also add `Settings.defaultZone = 'America/Los_Angeles'` to this file? How does `DateTime` now which timezone is local?

File: tests/utils/formatDate.test.ts [hunk 4]
```diff
@@ -88,3 +95,28 @@ describe('Formatting dates', () => {
...
+    const baseDate = DateTime.fromISO('2021-09-10T02:00:00')
+    const adjustment = -1
+    const newDate = formatFromApiGateway(adjustment, baseDate)
+
+    expect(newDate).toEqual('2021-09-09T00:00:00')
+  })
+
+  it('handles PT edge cases', () => {
+    // Tricky time - 10pm PT means its always* the next day ET & UTC
+    // *like, casually true but don't hold me to this time is hard okay?
```
Comment (rocketnova) [inline tests/utils/formatDate.test.ts:115]:
```suggestion
    // Tricky time - 10pm PT means it's always* the next day ET & UTC
    // *like, casually true but don't hold me to this; time is hard okay?
```

File: tests/utils/formatDate.test.ts [hunk 5]
```diff
@@ -1,7 +1,14 @@
 import MockDate from 'mockdate'
 import { toDate } from 'date-fns-tz'
+import { DateTime, Settings } from 'luxon'
```
Comment (rocketnova) [inline tests/utils/formatDate.test.ts:3]: I'm getting a linting warning that `Settings` is imported but not used

**Review by rocketnova — APPROVED** [review]
A couple small questions. This looks good and we should 🚢 , but I think we should test Scenario 2 against dev on 8/25 to ensure it's working as expected on staging. I'll add a reminder on my calendar for next week

File: utils/formatDate.ts (hunk 1 above)
Comment (lomky) [reply utils/formatDate.ts:26]: good nit - but we end up not using the `pacificTimeZone` anywhere else so I just delete it in the follow up!

File: tests/utils/formatDate.test.ts (hunk 5 above)
Comment (lomky) [reply tests/utils/formatDate.test.ts:3]: Yup! 😅 cleaned up in next PR!

File: tests/utils/formatDate.test.ts (hunk 3 above)
Comment (lomky) [reply tests/utils/formatDate.test.ts:102]: Good question! I tested adding it in various places. Initially I was hopeful adding it to the ServerSide function in `page/index.tsx` would be sufficient, but no. Then I added it to here, also didn't work, and then I added it to utils, and it does work. I believe the Setting affects the behavior everywhere once its set

I think if I were more JS-Server-Stuff savvy I would know a better place to put it 🤔

Comment (rocketnova) [reply tests/utils/formatDate.test.ts:102]: I feel a little surprised by that behavior 🤔  but 🤷

**Comment by lomky** [comment]
Nooo idea why the locale isn't being respected on the CI here - it works locally 🤔 Feel free to debug while I'm gone, otherwise I'll pick up Thursday!
